internal/service/ec2/vpc_security_group.go

```diff
@@ -730,7 +730,7 @@ func resourceSecurityGroupUpdateRules(
 						GroupId:       group.GroupId,
 						IpPermissions: remove,
 					}
-					if group.VpcId == nil || *group.VpcId == "" {
+					if aws.StringValue(group.VpcId) == "" {
```

Comment:
For my understanding, are the nil checks no longer needed when using `aws.StringValue()` ?

Reply:
Correct, `aws.XYZValue()` converts a `nil` into the type's zero value (`""` for `string`).